mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraApplicableFlagDAO.java

```diff
@@ -75,36 +71,26 @@ private PreparedStatement prepareSelect(Session session) {
                 rowOptional.map(row -> new FlagsExtractor(row).getApplicableFlags()));
     }
 
-    public CompletableFuture<Void> updateApplicableFlags(CassandraId cassandraId, Flags flags) {
-        Flags newFlags = new FlagsBuilder().add(flags).build();
-        newFlags.remove(Flag.RECENT);
-        newFlags.remove(Flag.USER);
+    public CompletableFuture<Void> updateApplicableFlags(CassandraId cassandraId, Flags oldFlags, Flags newFlags) {
+        ImmutableSet<String> oldUserFlags = ImmutableSet.copyOf(oldFlags.getUserFlags());
+        ImmutableSet<String> newUserFlags = ImmutableSet.copyOf(newFlags.getUserFlags());
+        Sets.SetView<String> addedUserFlags = Sets.difference(newUserFlags, oldUserFlags);
 
-        if (newFlags.equals(EMPTY_FLAGS)) {
+        if (addedUserFlags.isEmpty()) {
```

Comment:
Yes.

Applicable flags means "flags that have ever been used with this mailbox"

Thus we don't need to handle removed flags.

It's by the way perfectly specified in the unit tests of a previous PR @quynhn  handled.